### ISSUE TITLE
Cannot build YouCompleteMe on Darwin

### ISSUE THREAD

**Issue by anderslundstedt** [NixOS/nixpkgs#16212]
## Issue description

Building `nixpkgs.vimPlugins.YouCompleteMe` on Darwin fails.
### Steps to reproduce

`nix-env -iA nixpkgs.vimPlugins.YouCompleteMe`
### Related issues

The problem seems very similar to #8880 which was solved by #8891—however I do not know how to apply that solution here.
### Build output

```
installing ‘vimplugin-youcompleteme-2016-05-17’
these derivations will be built:
  /nix/store/f3bss4vckd4qccq4dsaiv0fab4qk1gyn-vimplugin-youcompleteme-2016-05-17.drv
building path(s) ‘/nix/store/zxgzrbvfigf9vm5l81g559pm2b6843zv-vimplugin-youcompleteme-2016-05-17’
unpacking sources
unpacking source archive /nix/store/xwzplcvb3s3ihf7i70qn2vrgxp3kbvwa-youcompleteme-fa58393
source root is youcompleteme-fa58393
patching sources
configuring
building
patching script interpreter paths in .
./install.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./install.sh: interpreter directive changed from "/bin/sh" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/sh  "
./print_todos.sh: interpreter directive changed from "/bin/bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash  "
./run_tests.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/pythonfutures/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/build.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/fuse_gtest_files.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/gen_gtest_pred_impl.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/gtest-config.in: interpreter directive changed from "/bin/sh" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/sh  "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/pump.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/release_docs.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/upload.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/scripts/upload_gtest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_break_on_failure_unittest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_catch_exceptions_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_color_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_env_var_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_filter_unittest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_help_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_list_tests_unittest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_output_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_shuffle_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_test_utils.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_throw_on_failure_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_uninitialized_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_xml_outfiles_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_xml_output_unittest.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/gtest/test/gtest_xml_test_utils.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/fuse_gmock_files.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/ast.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/gmock_class.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/gmock_class_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/keywords.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/tokenize.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/cpp/utils.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/generator/gmock_gen.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/gmock-config.in: interpreter directive changed from "/bin/sh" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/sh  "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/gmock_doctor.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/upload.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/scripts/upload_gmock.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/test/gmock_leak_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/test/gmock_output_test.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/cpp/ycm/tests/gmock/test/gmock_test_utils.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/examples/example_client.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/run_tests.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/style_format.sh: interpreter directive changed from "/bin/bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash  "
./third_party/ycmd/third_party/bottle/docs/_locale/update.sh: interpreter directive changed from "/bin/bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash  "
./third_party/ycmd/third_party/bottle/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/bottle/test/testall.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/bottle/test/travis_setup.sh: interpreter directive changed from "/bin/bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash  "
./third_party/ycmd/third_party/gocode/_testing/all.bash: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash "
./third_party/ycmd/third_party/gocode/_testing/run.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/gocode/debian/rules: interpreter directive changed from "/usr/bin/make -f" to "/nix/store/fgciyn787jdr8gkha6qdm3fsnxw0a0hr-gnumake-4.1/bin/make -f "
./third_party/ycmd/third_party/gocode/vim/pathogen_update.sh: interpreter directive changed from "/bin/sh" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/sh  "
./third_party/ycmd/third_party/gocode/vim/symlink.sh: interpreter directive changed from "/bin/sh" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/sh  "
./third_party/ycmd/third_party/gocode/vim/update.sh: interpreter directive changed from "/bin/sh" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/sh  "
./third_party/ycmd/third_party/JediHTTP/vendor/bottle/docs/_locale/update.sh: interpreter directive changed from "/bin/bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash  "
./third_party/ycmd/third_party/JediHTTP/vendor/bottle/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/JediHTTP/vendor/bottle/test/testall.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/JediHTTP/vendor/bottle/test/travis_setup.sh: interpreter directive changed from "/bin/bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash  "
./third_party/ycmd/third_party/JediHTTP/vendor/jedi/scripts/memory_check.py: interpreter directive changed from " /usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/JediHTTP/vendor/jedi/scripts/profile.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/JediHTTP/vendor/jedi/scripts/wx_check.py: interpreter directive changed from " /usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/JediHTTP/vendor/jedi/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/JediHTTP/vendor/jedi/sith.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/JediHTTP/vendor/jedi/test/refactor.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/JediHTTP/vendor/jedi/test/run.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/python-future/check_rst.sh: interpreter directive changed from "/bin/sh" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/sh  "
./third_party/ycmd/third_party/python-future/futurize.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/python-future/pasteurize.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/python-future/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/racerd/scripts/assert_cargo_lock_unchanged: interpreter directive changed from "/bin/bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash  "
./third_party/ycmd/third_party/racerd/scripts/download_rust_source.sh: interpreter directive changed from "/bin/sh" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/sh  "
./third_party/ycmd/third_party/racerd/scripts/run_wrk_benchmarks.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/requests/requests/packages/chardet/chardetect.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/requests/setup.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/third_party/requests/test_requests.py: interpreter directive changed from "/usr/bin/env python" to "/nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python "
./third_party/ycmd/update_boost.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/8ph1cmfy27yhvcifbdqm04g3dxbgswbs-bash-4.3-p42/bin/bash "
/private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/build /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393
-- The C compiler identification is Clang 3.8.0
-- The CXX compiler identification is Clang 3.8.0
-- Check for working C compiler: /nix/store/m04drk29xxv9lxrl9qbs9y1abkglnwbm-clang-wrapper-3.8.0/bin/clang
-- Check for working C compiler: /nix/store/m04drk29xxv9lxrl9qbs9y1abkglnwbm-clang-wrapper-3.8.0/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /nix/store/m04drk29xxv9lxrl9qbs9y1abkglnwbm-clang-wrapper-3.8.0/bin/clang++
-- Check for working CXX compiler: /nix/store/m04drk29xxv9lxrl9qbs9y1abkglnwbm-clang-wrapper-3.8.0/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
Your C++ compiler supports C++11, compiling in that mode.
-- Found PythonLibs: /nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/lib/libpython2.7.dylib (found suitable version "2.7.11", minimum required is "2.6")
Using libclang to provide semantic completion for C/C++/ObjC
Using external libclang: /nix/store/7nznkn43afvb0axjd6d3136mcjllwzhh-clang-3.8.0/lib/libclang.dylib
-- Found PythonInterp: /nix/store/r665msgxzc9rpl7y7xpikidbq29zdxvd-python-2.7.11/bin/python2.7 (found version "2.7.11")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/build
Scanning dependencies of target BoostParts
[  0%] Building CXX object BoostParts/CMakeFiles/BoostParts.dir/libs/atomic/src/lockpool.cpp.o
[  0%] Building CXX object BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/process_cpu_clocks.cpp.o
[  1%] Building CXX object BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/chrono.cpp.o
[  2%] Building CXX object BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/thread_clock.cpp.o
[  3%] Building CXX object BoostParts/CMakeFiles/BoostParts.dir/libs/date_time/src/gregorian/date_generators.cpp.o
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/libs/date_time/src/gregorian/date_generators.cpp:14:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/date_time/date_generators.hpp:17:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/sstream:174:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/ostream:138:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/ios:216:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/__locale:15:
/nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/string:1938:44: error: 'basic_string<_CharT, _Traits, _Allocator>' is missing exception specification 'noexcept(is_nothrow_copy_constructible<allocator_type>::value)'
basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
                                           ^
/nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/string:1326:40: note: previous declaration is here
    _LIBCPP_INLINE_VISIBILITY explicit basic_string(const allocator_type& __a)
                                       ^
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/libs/chrono/src/process_cpu_clocks.cpp:17:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/detail/inlined/process_cpu_clocks.hpp:19:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/process_cpu_clocks.hpp:20:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/time_point.hpp:34:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/iostream:38:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/ios:216:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/__locale:15:
/nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/string:1938:44: error: 'basic_string<_CharT, _Traits, _Allocator>' is missing exception specification 'noexcept(is_nothrow_copy_constructible<allocator_type>::value)'
basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
                                           ^
/nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/string:1326:40: note: previous declaration is here
    _LIBCPP_INLINE_VISIBILITY explicit basic_string(const allocator_type& __a)
                                       ^
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/libs/chrono/src/chrono.cpp:14:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/detail/inlined/chrono.hpp:13:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/chrono.hpp:12:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/time_point.hpp:34:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/iostream:38:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/ios:216:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/__locale:15:
/nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/string:1938:44: error: 'basic_string<_CharT, _Traits, _Allocator>' is missing exception specification 'noexcept(is_nothrow_copy_constructible<allocator_type>::value)'
basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
                                           ^
/nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/string:1326:40: note: previous declaration is here
    _LIBCPP_INLINE_VISIBILITY explicit basic_string(const allocator_type& __a)
                                       ^
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/libs/chrono/src/thread_clock.cpp:18:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/detail/inlined/thread_clock.hpp:17:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/thread_clock.hpp:19:
In file included from /private/var/folders/sb/6h097wtn38z0vcx6mxbrl49w0000gn/T/nix-build-vimplugin-youcompleteme-2016-05-17.drv-0/youcompleteme-fa58393/third_party/ycmd/cpp/BoostParts/boost/chrono/time_point.hpp:34:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/iostream:38:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/ios:216:
In file included from /nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/__locale:15:
/nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/string:1938:44: error: 'basic_string<_CharT, _Traits, _Allocator>' is missing exception specification 'noexcept(is_nothrow_copy_constructible<allocator_type>::value)'
basic_string<_CharT, _Traits, _Allocator>::basic_string(const allocator_type& __a)
                                           ^
/nix/store/yxpwamjdapjcp53mmsdh1j2c9bc26h4k-libc++-3.7.1/include/c++/v1/string:1326:40: note: previous declaration is here
    _LIBCPP_INLINE_VISIBILITY explicit basic_string(const allocator_type& __a)
                                       ^
1 error generated.
BoostParts/CMakeFiles/BoostParts.dir/build.make:158: recipe for target 'BoostParts/CMakeFiles/BoostParts.dir/libs/date_time/src/gregorian/date_generators.cpp.o' failed
make[3]: *** [BoostParts/CMakeFiles/BoostParts.dir/libs/date_time/src/gregorian/date_generators.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
1 error generated.
BoostParts/CMakeFiles/BoostParts.dir/build.make:110: recipe for target 'BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/process_cpu_clocks.cpp.o' failed
make[3]: *** [BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/process_cpu_clocks.cpp.o] Error 1
1 error generated.
1 error generated.
BoostParts/CMakeFiles/BoostParts.dir/build.make:86: recipe for target 'BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/chrono.cpp.o' failed
make[3]: *** [BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/chrono.cpp.o] Error 1
BoostParts/CMakeFiles/BoostParts.dir/build.make:134: recipe for target 'BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/thread_clock.cpp.o' failed
make[3]: *** [BoostParts/CMakeFiles/BoostParts.dir/libs/chrono/src/thread_clock.cpp.o] Error 1
CMakeFiles/Makefile2:85: recipe for target 'BoostParts/CMakeFiles/BoostParts.dir/all' failed
make[2]: *** [BoostParts/CMakeFiles/BoostParts.dir/all] Error 2
CMakeFiles/Makefile2:154: recipe for target 'ycm/CMakeFiles/ycm_core.dir/rule' failed
make[1]: *** [ycm/CMakeFiles/ycm_core.dir/rule] Error 2
Makefile:131: recipe for target 'ycm_core' failed
make: *** [ycm_core] Error 2
builder for ‘/nix/store/f3bss4vckd4qccq4dsaiv0fab4qk1gyn-vimplugin-youcompleteme-2016-05-17.drv’ failed with exit code 2
error: build of ‘/nix/store/f3bss4vckd4qccq4dsaiv0fab4qk1gyn-vimplugin-youcompleteme-2016-05-17.drv’ failed
```

**Comment by vcunat** [comment]
Wasn't this fixed by merging https://github.com/NixOS/nixpkgs/pull/16830?

**Comment by LnL7** [comment]
This issue is fixed, but it broke again recently.

**Comment by veprbl** [comment]
Proposed fix #21541

**Comment by LnL7** [comment]
Fixed by https://github.com/NixOS/nixpkgs/pull/21574
